### ISSUE TITLE
Add a delete button and a progress meter for Downloads

### ISSUE THREAD

**Issue by Numerosityy** [brave/brave-browser#35432]
These are just things I loved about Edge that I kinda miss when using Brave.

Sometimes I click twice cause I don't think the download has started, when in reality I've downloaded it twice. It would be nice if I could just delete the file from the Brave browser when it's finished instead of having to open File Explorer.

It would also be nice to be able to see the progress of a download through a meter. It would also show the download speed, that way I know whether I should try an alternative download link if it's too slow.

**Comment by rebron** [comment]
<img width="416" alt="Screenshot 2024-01-19 at 12 44 13 PM" src="https://github.com/brave/brave-browser/assets/5951041/3fd949a7-df76-4d81-bac7-596d6ce9d3e3">

There's already a progress meter for downloads as well as an indicator on the icon. You can right click to remove. We can look into adding a button for deletion and showing download speed though.

**Comment by rebron** [comment]
cc: @aguscruiz and @simonhong for thoughts. This would be to show a delete icon similar to Edge as well as showing the speed/rate of download whereas we show progress and state.

**Comment by aguscruiz** [comment]
I agree we should bring back the download speed back. Didn't realize we didn't have it anymore.

We could also add a shortcut for downloading, sure, let's see how it would work

**Comment by zeusshomaly** [comment]
About time! Looking forward to this!

**Comment by aguscruiz** [comment]
Designs available here:
https://www.figma.com/design/tU9zrk9Ite8B06JSjHAHhn/General-dialogs-and-popups?node-id=488-5

**Comment by zeusshomaly** [comment]
> Designs available here: https://www.figma.com/design/tU9zrk9Ite8B06JSjHAHhn/General-dialogs-and-popups?node-id=488-5

That is lovely of you @aguscruiz but I refuse to pay for figma, please send a screenshot, thank you :) if not then all good, it's okay. 

At the very least, confirm if you guys are still working on it thanks :)

Kind regards,
Zeus (aged 19) from Australia

**Comment by aguscruiz** [comment]
Sure thing, here are some screenshots: 

![Image](https://github.com/user-attachments/assets/81e73bf8-9e4b-4035-8a99-042db3c0137b)

![Image](https://github.com/user-attachments/assets/8bf8cbb1-c1b1-41fe-9736-5adfa8ac93ae)

**Comment by zeusshomaly** [comment]
> Sure thing, here are some screenshots:
> 
> ![Image](https://github.com/user-attachments/assets/81e73bf8-9e4b-4035-8a99-042db3c0137b)
> 
> ![Image](https://github.com/user-attachments/assets/8bf8cbb1-c1b1-41fe-9736-5adfa8ac93ae)

@aguscruiz You are truly a legend buddy. I truly appreciate this! Thank you so much! This is perfect! Let me know when it comes out or if it's still in development, I am so excited!!!

**Comment by zeusshomaly** [comment]
@aguscruiz thank you again for the screenshots, I just created a brave community post regarding it with your screenshot, tysm!

https://community.brave.com/t/a-delete-file-button-in-the-downloads-bar-like-edge/625281

**Comment by aguscruiz** [comment]
Great, thanks! :D

**Comment by zeusshomaly** [comment]
You're a real one for that! Oh and I also hear you guys are working on upgrading the sidebar to be like Edge's that opens stuff to the side and not in a new tab, as @simonhong is working actively on! He or someone did add a design of that via figma, but again I refuse to pay. if you can, if not no problem, happy to look for that and send me a screenshot? thanks again @aguscruiz 

Link to it: https://www.figma.com/design/Dt0iJdVaaAj7DikWLO1P7I/Split-view-%2F-Sidebar-browsing?node-id=2695-36684&p=f&m=dev

This will help me a lot! As i'm excited for the sidebar too!

**Comment by zeusshomaly** [comment]
@aguscruiz My pleasure! I hope the progress of it's development is well going, 5 days later :)

I am looking forward for it's release.

**Comment by zeusshomaly** [comment]
Thanks @rebron I am very excited for this feature!

**Comment by zeusshomaly** [comment]
Woo hoo @sangwoo108 , you did it! @sangwoo108 I would like to say a massive thank you for all this! I am looking forward for it's release, no hurry! You sir are my hero! You said earlier it was a bit complicated, but now you managed to complete it. I am extremely looking forward for this delete button and the customise toolbar. You are a legend!

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using 
```
Brave | 1.82.114 Chromium: 139.0.7258.42 (Official Build) nightly (64-bit)
-- | --
Revision | 141a3655ce8e372908eceb30ab3425a2a2ed4f34
OS | Windows 11 Version 24H2 (Build 26100.4652)
```

1. Installed 1.82.114
2. launched Brave
3. navigated to couple of sites to download files 
  - https://github.com/brave/brave-browser/releases/tag/v1.82.116
  - https://ubuntu.com/download/desktop/thank-you?version=24.04.2&architecture=amd64&lts=true
4. waited until download is complete
5. opened download bubble in the toolbar

#### Confirmed delete button is shown in the download bubble, and when clicked, downloaded file is removed  

https://github.com/user-attachments/assets/6e75dc64-43d8-4103-abf2-72343807b999
